### PR TITLE
NET-125: ESLint rules

### PR DESCRIPTION
**Pull request by teogeb**
Fine-tuned the rule set: added TypeScript specific rules and removed redundant rules. Some rules are used directly instead of extending streamr-nodejs base rule set.

### CODE REVIEW

File: src/connection/WebRtcEndpoint.ts [hunk 1]
```diff
@@ -3,17 +3,17 @@ import nodeDataChannel, { DescriptionType } from 'node-datachannel'
 import getLogger from '../helpers/logger'
 import { PeerInfo } from './PeerInfo'
 import { Connection } from './Connection'
-import { Metrics, MetricsContext } from "../helpers/MetricsContext";
+import { Metrics, MetricsContext } from '../helpers/MetricsContext';
```
Comment (harbu) [inline src/connection/WebRtcEndpoint.ts]: Hmmm semicolon? Shouldn't linter fail?

File: src/helpers/SeenButNotPropagatedSet.ts [hunk 2]
```diff
@@ -1,4 +1,4 @@
-import LRUCache from "lru-cache";
+import LRUCache from 'lru-cache';
```
Comment (harbu) [inline src/helpers/SeenButNotPropagatedSet.ts]: Semicolon

**Review by harbu — APPROVED** [review]
LGTM; semicolons not getting detected?

File: src/connection/WebRtcEndpoint.ts (hunk 1 above)
Comment (teogeb) [reply src/connection/WebRtcEndpoint.ts]: Fixed

**Pull request merged**